components/validator/l2_output_submitter.go

```diff
@@ -149,49 +148,70 @@ func (l *L2OutputSubmitter) retryAfter(d time.Duration) {
 	})
 }
 
-// TODO(seolaoh): return wait duration explicitly, and handle `retryAfter` function calls at once.
+// trySubmitL2Output l2 Output will continue to check for submittable conditions and try.
+// If it needs to wait, it will check the condition again after the delay.
```

Comment:
```suggestion
// trySubmitL2Output checks if the validator can submit l2 output and tries to submit it.
// If it needs to wait, it will calculate how long the validator should wait and try again after the delay.
```